atc/db/volume_repository.go

```diff
@@ -540,6 +542,53 @@ func (repository *volumeRepository) GetDestroyingVolumes(workerName string) ([]s
 	)
 }
 
+func (repository *volumeRepository) DestroyUnknownVolumes(workerName string, reportedHandles []string) (int, error) {
+	dbHandles, err := repository.queryVolumeHandles(sq.Eq{
+		"worker_name": workerName,
+	})
+	if err != nil {
+		return 0, err
+	}
+
+	unknownHandles := diff(reportedHandles, dbHandles)
+
+	if len(unknownHandles) == 0 {
+		return 0, nil
+	}
+
+	tx, err := repository.conn.Begin()
+	if err != nil {
+		return 0, err
+	}
+
+	defer Rollback(tx)
+
+	for _, unknownHandle := range unknownHandles {
+		_, err = psql.Insert("volumes").
```

Comment:
Do we want to do the same with the destroying container inserts where we do one large insert rather than a ton of single row inserts?

Reply:
![image](https://user-images.githubusercontent.com/19290703/66848384-fc393b80-ef42-11e9-96ab-087c1f898fb0.png)